synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -200,6 +200,10 @@ def __init__(self, hs, room_id, rules_for_room_cache):
         # not update the cache with it.
         self.sequence = 0
 
+        # A cache of user_ids that we *know* aren't interesting, e.g. user_ids
+        # owned by AS's, or remote users, etc.
+        self.uninteresting_user_set = set()
```

Comment:
Also might be worth explaining that "uninteresting" means "will never have push rules on this server"